internal/workflow.go

```diff
@@ -46,92 +46,229 @@ var (
+		// it), otherwise it returns false.
+		//
+		// This is equivalent to:
+		//  select {
+		//  case aChannel <- v: ok = true
+		//  default: ok = false
+		//  }
 		SendAsync(v interface{}) (ok bool)
 
-		// Close close the Channel, and prohibit subsequent sends.
+		// Close closes the Channel, and prohibits subsequent sends.
+		// As with a normal Go channel that has been closed, sending to a closed channel will panic.
 		Close()
 	}
 
-	// Selector must be used instead of native go select by workflow code for determinism.
-	// Use workflow.NewSelector(ctx) method to create a Selector instance.
-	// The interface is to simulate Golang's Select statement.
-	// For example, the logic of Golang code like below
-	// 	chA := make(chan int)
-	//	chB := make(chan int)
-	//	counter := 0
-	//	for {
-	//		select {
-	//		case i, ok := <- chA:
-	//			if ok{
-	//				counter += i
-	//			}
-	//		case i, ok := <- chB:
-	//			if ok{
-	//				counter += i
-	//			}
-	//		}
-	//	}
-	// should be written as
-	// 	s := workflow.NewSelector(ctx)
-	//	counter := 0
-	//	s.AddReceive(workflow.GetSignalChannel(ctx, "channelA"), func(c workflow.Channel, ok bool) {
-	//		if ok{
-	//			var i int
-	//			c.Receive(ctx, &i)
-	//			counter += i
-	//		}
-	//	})
-	//	s.AddReceive(workflow.GetSignalChannel(ctx, "channelB"), func(c workflow.Channel, ok bool) {
-	//		if ok{
-	//			var i int
-	//			c.Receive(ctx, &i)
-	//			counter += i
-	//		}
-	//	})
+	// Selector must be used in workflows instead of a native Go select statement.
+	//
+	// Use workflow.NewSelector(ctx) to create a Selector instance, and then add cases to it with its methods.
+	// The interface is intended to simulate Go's select statement, and any Go select can be fairly trivially rewritten
+	// for a Selector with effectively identical behavior.
+	//
+	// For example, normal Go code like below:
+	//  chA := make(chan int)
+	//  chB := make(chan int)
+	//  counter := 0
+	//  for {
+	//  	select {
+	//  	case i, more := <- chA:
+	//  		if more {
+	//  			counter += i
+	//  		}
+	//  	case i, more := <- chB:
+	//  		if more {
+	//  			counter += i
+	//  		}
+	//  	case <- time.After(time.Hour):
+	//  		break
+	//  	}
+	//  }
+	// can be written as:
+	//  chA := workflow.NewChannel(ctx)
+	//  chB := workflow.NewChannel(ctx)
+	//  counter := 0
+	//  for {
+	//  	timedout := false
+	//  	s := workflow.NewSelector(ctx)
+	//  	s.AddReceive(chA, func(c workflow.Channel, more bool) {
+	//  		if more {
+	//  			var i int
+	//  			c.Receive(ctx, &i)
+	//  			counter += i
+	//  		}
+	//  	})
+	//  	s.AddReceive(chB, func(c workflow.Channel, more bool) {
+	//  		if more {
+	//  			var i int
+	//  			c.Receive(ctx, &i)
+	//  			counter += i
+	//  		}
+	//  	})
+	//  	s.AddFuture(workflow.NewTimer(ctx, time.Hour), func(f workflow.Future) {
+	//  		timedout = true
+	//  	})
+	//  	s.Select(ctx)
+	//  	if timedout {
+	//  		break
+	//  	}
+	//  }
```

Comment:
Q: 
I guess there is no difference, but do you have any idea/preference? 

If we only keep the `s.Select(ctx)` in the for loop:
```golang
timedout := false
s := workflow.NewSelector(ctx)
s.AddReceive(chA, func(c workflow.Channel, more bool) {
	if more {
		var i int
		c.Receive(ctx, &i)
		counter += i
	}
})
s.AddReceive(chB, func(c workflow.Channel, more bool) {
	if more {
		var i int
		c.Receive(ctx, &i)
		counter += i
	}
})
s.AddFuture(workflow.NewTimer(ctx, time.Hour), func(f workflow.Future) {
	timedout = true
})

for {
  s.Select(ctx)
  if timedout {
	  break
  }
}
```

Reply:
for this one there is a difference - in the original with `case <-time.After`, the timer is restarted on each loop.  the same thing happens in the translated version.

I could pull the timer out separately (`after := time.After(time.Hour)` + `case <-after`).  semantically they're a bit different (one waits for N things within 1 hour, the other waits for N things _each_ hour), no real preference which we demonstrate.
but I do want a timeout of some kind in one of these docs chunks, since it's not the most obvious part of the API, and timeouts are kinda common.

Reply:
on second (or third or whatever) thought: I should probably add a comment about intended behavior anyway.  no need to leave people to puzzle it out (timeout vs idle-timeout) while reading about more relevant things.